packages/apps/reputation-oracle/server/src/modules/reputation/reputation.service.ts

```diff
@@ -1,44 +1,166 @@
-import { Injectable, Logger, NotFoundException } from '@nestjs/common';
+import { Inject, Injectable, Logger, NotFoundException } from '@nestjs/common';
 import { ConfigService } from '@nestjs/config';
 import { ChainId } from '@human-protocol/sdk';
 import { INITIAL_REPUTATION } from '../../common/constants';
 import { ConfigNames } from '../../common/config';
-import { ReputationEntityType, ReputationLevel } from '../../common/enums';
+import {
+  JobRequestType,
+  ReputationEntityType,
+  ReputationLevel,
+  SolutionError,
+} from '../../common/enums';
 import { ReputationRepository } from './reputation.repository';
-import { ErrorReputation } from '../../common/constants/errors';
+import {
+  ErrorManifest,
+  ErrorReputation,
+  ErrorResults,
+} from '../../common/constants/errors';
 import { ReputationDto } from './reputation.dto';
+import { StorageService } from '../storage/storage.service';
+import { Web3Service } from '../web3/web3.service';
+import { EscrowClient } from '@human-protocol/sdk';
+import { CvatManifestDto, FortuneManifestDto } from '../../common/dto/manifest';
+import { FortuneFinalResult } from '../../common/dto/result';
 
 @Injectable()
 export class ReputationService {
   private readonly logger = new Logger(ReputationService.name);
 
   constructor(
+    @Inject(StorageService)
+    private readonly storageService: StorageService,
     private readonly reputationRepository: ReputationRepository,
     private readonly configService: ConfigService,
+    private readonly web3Service: Web3Service,
   ) {}
 
-  public getReputationLevel(reputationPoints: number): ReputationLevel {
-    const reputationLevelLow = this.configService.get<number>(
-      ConfigNames.REPUTATION_LEVEL_LOW,
-      300,
-    );
+  /**
+   * Perform reputation assessment based on the completion status of a job and its associated entities.
+   * Retrieves necessary data from the escrow client, including manifest and final results URLs,
+   * and delegates reputation adjustments to specialized methods.
+   * @param chainId The ID of the blockchain chain.
+   * @param escrowAddress The address of the escrow contract.
+   * @param checkPassed A boolean indicating if the job completion check passed.
+   * @returns {Promise<void>} A Promise indicating the completion of reputation assessment.
+   */
+  public async doAssessReputationScores(
```

Comment:
Here we could do something like [this](https://github.com/humanprotocol/human-protocol/pull/1574/files), so we have a specific assessment for each job type.